homeassistant/components/webostv/media_player.py

```diff
@@ -326,46 +282,54 @@ def supported_features(self):
             return SUPPORT_WEBOSTV | SUPPORT_TURN_ON
         return SUPPORT_WEBOSTV
 
-    def turn_off(self):
+    @cmd
+    async def async_turn_off(self):
         """Turn off media player."""
+        await self._client.power_off()
 
-        self._state = STATE_OFF
-        try:
-            self._client.power_off()
-        except (OSError, ConnectionClosed, TypeError, asyncio.TimeoutError):
-            pass
-
-    def turn_on(self):
+    async def async_turn_on(self):
         """Turn on the media player."""
+        connected = self._client.is_connected()
         if self._on_script:
-            self._on_script.run()
+            await self._on_script.async_run()
+
+        # if connection was already active
+        # ensure is still alive
+        if connected:
+            await self._client.get_current_app()
 
-    def volume_up(self):
+    @cmd
+    async def async_volume_up(self):
         """Volume up the media player."""
-        self._client.volume_up()
+        await self._client.volume_up()
 
-    def volume_down(self):
+    @cmd
+    async def async_volume_down(self):
         """Volume down media player."""
-        self._client.volume_down()
+        await self._client.volume_down()
 
-    def set_volume_level(self, volume):
+    @cmd
+    def async_set_volume_level(self, volume):
         """Set volume level, range 0..1."""
         tv_volume = volume * 100
-        self._client.set_volume(tv_volume)
+        return self._client.set_volume(tv_volume)
```

Comment:
Nothing is checking this return value.

Reply:
In this case it's returning the coroutine.

Reply:
This method, `async_set_volume_level` should be a coroutine and we should await `self._client.set_volume(tv_volume)`.